context-free.md

```diff
@@ -2,21 +2,42 @@
 
 ##  Who is this programming language for?
 
+This programming language is for people who want to create art on the computer 
+through rules. Instead of using a mouse to point and click like you would in 
+paint, you rather create mathematical rules that creates art.
 
 ## What is easy to do in this language? Why is it easy?
 
+It is easy to do recursive shapes. It is easy because the language allows you 
+to define rules with itself.
 
 ## What is hard to do in this language? Why is it hard?
 
+It is hard to do arbitrary shapes. If you wanted to draw letters, you have to 
+define multiple different and somewhat repetitive rules. It's nice to have the 
+render button to show you what you did immediately, but complex arbitrary 
+shapes still take a while. In the end this is because the rules do not cater 
+to this.
 
 ## How did you learn how to program in this language?
 
+I learned how to program in this language by looking at examples and searching 
+for online tutorials.
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+The underlying computational model of ContextFree is to render shapes as 
+they are defined in their own rules in the program.
+
```

Comment:
I think this answer can be more elaborate on how it renders shapes from the user-defined rules. In my answer, I mentioned that it stops computing recursive rules when they reach a state where there's only primitive shapes, which is the base case.